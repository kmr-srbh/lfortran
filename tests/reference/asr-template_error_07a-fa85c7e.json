{
    "basename": "asr-template_error_07a-fa85c7e",
    "cmd": "lfortran --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/errors/template_error_07a.f90",
    "infile_hash": "2d93392c48b94ce432551da534053cf051a4d2ff3aa403b134d4a8e8",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-template_error_07a-fa85c7e.stdout",
    "stdout_hash": "8b4c721ad13afd062ea89204c7de76e28b146725545deb458d2de2a2",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}