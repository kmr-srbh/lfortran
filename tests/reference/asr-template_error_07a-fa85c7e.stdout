(TranslationUnit
    (SymbolTable
        1
        {
            template_01:
                (Program
                    (SymbolTable
                        6
                        {
                            semigroup:
                                (ExternalSymbol
                                    6
                                    semigroup
                                    2 semigroup
                                    template_01_m
                                    []
                                    semigroup
                                    Public
                                ),
                            test_template:
                                (ExternalSymbol
                                    6
                                    test_template
                                    2 test_template
                                    template_01_m
                                    []
                                    test_template
                                    Public
                                )
                        })
                    template_01
                    [template_01_m]
                    []
                ),
            template_01_m:
                (Module
                    (SymbolTable
                        2
                        {
                            semigroup:
                                (Requirement
                                    (SymbolTable
                                        3
                                        {
                                            combine:
                                                (Function
                                                    (SymbolTable
                                                        4
                                                        {
                                                            combined:
                                                                (Variable
                                                                    4
                                                                    combined
                                                                    []
                                                                    ReturnVar
                                                                    ()
                                                                    ()
                                                                    Default
                                                                    (TypeParameter
                                                                        t
                                                                    )
                                                                    ()
                                                                    Source
                                                                    Private
                                                                    Required
                                                                    .false.
                                                                ),
                                                            x:
                                                                (Variable
                                                                    4
                                                                    x
                                                                    []
                                                                    In
                                                                    ()
                                                                    ()
                                                                    Default
                                                                    (TypeParameter
                                                                        t
                                                                    )
                                                                    ()
                                                                    Source
                                                                    Private
                                                                    Required
                                                                    .false.
                                                                ),
                                                            y:
                                                                (Variable
                                                                    4
                                                                    y
                                                                    []
                                                                    In
                                                                    ()
                                                                    ()
                                                                    Default
                                                                    (TypeParameter
                                                                        t
                                                                    )
                                                                    ()
                                                                    Source
                                                                    Private
                                                                    Required
                                                                    .false.
                                                                )
                                                        })
                                                    combine
                                                    (FunctionType
                                                        [(TypeParameter
                                                            t
                                                        )
                                                        (TypeParameter
                                                            t
                                                        )]
                                                        (TypeParameter
                                                            t
                                                        )
                                                        Source
                                                        Implementation
                                                        ()
                                                        .true.
                                                        .false.
                                                        .false.
                                                        .false.
                                                        .false.
                                                        []
                                                        .true.
                                                    )
                                                    []
                                                    [(Var 4 x)
                                                    (Var 4 y)]
                                                    []
                                                    (Var 4 combined)
                                                    Private
                                                    .false.
                                                    .false.
                                                    ()
                                                ),
                                            t:
                                                (Variable
                                                    3
                                                    t
                                                    []
                                                    In
                                                    ()
                                                    ()
                                                    Default
                                                    (TypeParameter
                                                        t
                                                    )
                                                    ()
                                                    Source
                                                    Private
                                                    Required
                                                    .false.
                                                )
                                        })
                                    semigroup
                                    [t]
                                    []
                                ),
                            test_template:
                                (Function
                                    (SymbolTable
                                        5
                                        {
                                            
                                        })
                                    test_template
                                    (FunctionType
                                        []
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        .false.
                                    )
                                    []
                                    []
                                    []
                                    ()
                                    Private
                                    .false.
                                    .false.
                                    ()
                                )
                        })
                    template_01_m
                    []
                    .false.
                    .false.
                )
        })
    []
)
